guide/src/class.md

```diff
@@ -211,6 +211,32 @@ Python::with_gil(|py| {
 });
 ```
 
+### frozen classes: Opting out of interior mutability
+
+As detailed abvoe, runtime borrow checking is currently enabled by default. But a class can opt of out it by declaring itself `frozen`. It can still use interior mutability via standard Rust types like `RefCell` or `Mutex`, but it is not bound to the implementation provided by PyO3 and can choose the most appropriate strategy on field-by-field basis.
+
+Classes which are `frozen` and also `Sync`, e.g. they do use `Mutex` but not `RefCell`, can also be accessed independently of the acquiring the GIL via the `PyCell::get` and `Py::get` methods:
+
+```rust
+use std::sync::atomic::{AtomicUsize, Ordering};
+# use pyo3::prelude::*;
+
+#[pyclass(frozen)]
+struct FrozenCounter {
+    value: AtomicUsize,
+}
+
+let cell  = Python::with_gil(|py| {
+    let counter = FrozenCounter { value: AtomicUsize::new(0) };
+
+    Py::new(py, counter).unwrap()
+});
+
+cell.get().value.fetch_add(1, Ordering::Relaxed);
```

Comment:
It might be helpful to add a type annotation here to show the user what's going on:

```suggestion
let py_frozen_counter: Py<FrozenCounter> = Python::with_gil(|py| {
    let counter = FrozenCounter { value: AtomicUsize::new(0) };

    Py::new(py, counter).unwrap()
});

py_frozen_counter.get().value.fetch_add(1, Ordering::Relaxed);
```